src/Raven.Server/Documents/PeriodicBackup/RavenFtpClient.cs

```diff
@@ -130,72 +143,126 @@ private void ExtractUrlAndDirectories(out string url, out List<string> dirs)
             var uri = new Uri(_url);
 
             dirs = uri.AbsolutePath.TrimStart('/').TrimEnd('/').Split("/").ToList();
-            var port = _port ?? (uri.Port > 0 ? uri.Port : DefaultFtpPort);
-            if (port < 1 || port > 65535)
-                throw new ArgumentException("Port number range: 1-65535");
 
-            url = $"{uri.Scheme}://{uri.Host}:{port}";
+            url = $"{uri.Scheme}://{uri.Host}";
         }
 
-        private FtpWebRequest CreateFtpWebRequest(string url, string method, bool keepAlive)
+        internal FtpClient CreateFtpClient(string url, bool keepAlive)
         {
-#pragma warning disable SYSLIB0014
-            var request = (FtpWebRequest)WebRequest.Create(new Uri(url));
-#pragma warning restore SYSLIB0014
-            request.Method = method;
-
-            request.Credentials = new NetworkCredential(_userName, _password);
-            request.EnableSsl = _useSsl;
+            var client = new FtpClient(url);
+            client.Config.ConnectTimeout = -1;
+            client.Credentials = new NetworkCredential(_userName, _password);
             if (_useSsl)
             {
                 var byteArray = Convert.FromBase64String(_certificateAsBase64);
 
                 try
                 {
-                    var x509Certificate = new X509Certificate(byteArray);
-                    request.ClientCertificates = new X509CertificateCollection(new[] { x509Certificate });
+                    var x509Certificate = new X509Certificate2(byteArray);
+                    client.Config.EncryptionMode = FtpEncryptionMode.Explicit;
+                    client.Config.ClientCertificates.Add(x509Certificate);
+                    if (_isTesting)
+                        client.Config.ValidateAnyCertificate = true;
+                    else
+                        client.ValidateCertificate += new FtpSslValidation(OnValidateCertificate);
                 }
                 catch (Exception e)
                 {
                     throw new ArgumentException($"This is not a valid certificate, file name: {_certificateFileName}", e);
                 }
             }
-                
-            request.UsePassive = true;
-            request.UseBinary = true;
-            request.KeepAlive = keepAlive;
 
-            return request;
+            client.Config.SocketKeepAlive = keepAlive;
+            if (_port != null)
+                client.Port = (int)_port;
+            else
+                client.Port = DefaultFtpPort;
+            client.Connect();
+            return client;
+        }
+
+        private static void OnValidateCertificate(BaseFtpClient control, FtpSslValidationEventArgs e)
+        {
+            if (e.PolicyErrors == SslPolicyErrors.None)
+            {
+                e.Accept = true;
+            }
+            else
+            {
+                throw new Exception($"{e.PolicyErrors}{Environment.NewLine}{e.Certificate}");
```

Comment:
IIRC, there is a AuthenticationException available, wouldn't it be better? Also some message would be nice for the user so it would be easier to nail down the problem?

Reply:
Done